popup.js

```diff
@@ -114,7 +161,82 @@ document.addEventListener('DOMContentLoaded', () => {
             copyToClipboard(newLinkInput.value);
         });
         newLinkContainer.appendChild(newCopyIcon);
-    
+
         customLinksContainer.appendChild(newLinkContainer);
-    }    
+
+        if (isNew) {
+            // Make the name field editable initially for new links
+            newNameSpan.focus();
+        }
+
+        // Add input event listeners for auto-save
+        newNameSpan.addEventListener('input', () => {
+            if (newNameSpan.textContent.trim() !== '') {
+                newNameSpan.classList.remove('empty');
+            } else {
+                newNameSpan.classList.add('empty');
+            }
+            if (autoSaveButton.classList.contains('auto-save-enabled')) saveAllLinks();
+        });
+        newLinkInput.addEventListener('input', () => {
+            if (autoSaveButton.classList.contains('auto-save-enabled')) saveAllLinks();
+        });
+
+        if (autoSaveButton.classList.contains('auto-save-enabled') && isNew) saveAllLinks();
+    }
+
+    function editCustomLink(linkContainer) {
```

Comment:
I like how the name for the url for custom link looks now, but I'd prefer they can be edited on the spot like previously instead of having to go through the action of clicking edit -> clicking the edit icon -> edit -> save.